integrations/oci/oci-secrets-mp-config-source/src/main/java/io/helidon/integrations/oci/secrets/mp/configsource/AbstractSecretBundleConfigSource.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.integrations.oci.secrets.mp.configsource;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import com.oracle.bmc.model.BmcException;
+import com.oracle.bmc.secrets.Secrets;
+import com.oracle.bmc.secrets.model.Base64SecretBundleContentDetails;
+import com.oracle.bmc.secrets.model.SecretBundleContentDetails;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * An {@link AutoCloseable} {@link ConfigSource} that retrieves configuration property values from an <a
+ * href="https://docs.oracle.com/en-us/iaas/Content/KeyManagement/Concepts/keyoverview.htm" target="_top">Oracle Cloud
+ * Infrastructure Vault</a>.
+ */
+@SuppressWarnings("try")
+class AbstractSecretBundleConfigSource implements AutoCloseable, ConfigSource {
```

Comment:
There is no doubt in my mind that we will need to implement Polling and/or Event in order to support reloading of certificates with associated new secrets that go along with them. I'll file a tracking issue for that follow-up work. See https://github.com/helidon-io/helidon/issues/7323